mono/metadata/runtime.c

```diff
@@ -58,7 +58,8 @@ fire_process_exit_event (MonoDomain *domain, gpointer user_data)
 	ERROR_DECL (error);
 	MonoObject *exc = NULL;
 	MonoMethod *method = mono_class_get_method_from_name (mono_defaults.appdomain_class, "RunProcessExit", -1);
-	g_assert (method);
+	if (!method)
+		return;
```

Comment:
why not just bump the corlib version?

Reply:
The runtime needs to be linker/bcl profile resistant.

Reply:
I might be missing something but I don't see how this is different to any of the other methods we access from the runtime, that's why we need to keep them via the linker descriptor xml.

Do you mean there are profiles where this method won't exist?

Btw. I just saw that the RunProcessExit() method should be moved before this: 😄
https://github.com/mono/mono/blob/e3cd4e90bfb28e6d599fbac7d6f2989ef5f6dbc0/mcs/class/corlib/System/AppDomain.cs#L1436

Reply:
It's on the linker files, but there's no downside on making the runtime more resilient on missing optional features.